kyo-stm/shared/src/main/scala/kyo/STM.scala

```diff
@@ -202,6 +137,81 @@ object STM:
 
     end run
 
+    private def commit[A, S](tid: Long, log: TRefLog, probe: Boolean = false)(using AllowUnsafe): Boolean =
```

Comment:
The overall logic here is still the same. I've refactored it into a new method that returns a `Boolean` indicating if the commit was successful, introduced `boundary`/`break` to replace the exception throwing, and add a `probe` parameter to indicate that the commit should only validate and not write values to refs